File: src/main/java/io/bindernews/thegrackle/power/SwoopPower.java
package io.bindernews.thegrackle.power;

import com.megacrit.cardcrawl.actions.common.RemoveSpecificPowerAction;
import com.megacrit.cardcrawl.core.AbstractCreature;
import com.megacrit.cardcrawl.powers.AbstractPower;
import io.bindernews.thegrackle.GrackleMod;
import io.bindernews.thegrackle.stance.StanceAloft;

import static io.bindernews.thegrackle.helper.ModInterop.iop;

public class SwoopPower extends BasePower {
    public static final String POWER_ID = GrackleMod.makeId(SwoopPower.class);

    public SwoopPower(AbstractCreature owner) {
        super(POWER_ID);
        setOwnerAmount(owner, -1);
        type = PowerType.BUFF;
        isTurnBased = true;
        loadRegion("anger");
        updateDescription();
    }

    public SwoopPower(AbstractCreature owner, int ignoredAmount) {
        this(owner);
    }

    @Override
    public void atEndOfTurn(boolean isPlayer) {
        addToBot(iop().changeStance(owner, StanceAloft.STANCE_ID));
        addToBot(new RemoveSpecificPowerAction(owner, owner, this));
    }

    @Override
    public AbstractPower makeCopy() {
        return new SwoopPower(owner);
    }
}

File: src/main/kotlin/io/bindernews/thegrackle/power/SwoopPower.kt
package io.bindernews.thegrackle.power

import com.megacrit.cardcrawl.actions.common.RemoveSpecificPowerAction
import com.megacrit.cardcrawl.core.AbstractCreature
import io.bindernews.thegrackle.GrackleMod
import io.bindernews.thegrackle.helper.ModInterop.Companion.iop
import io.bindernews.thegrackle.stance.StanceAloft

class SwoopPower(owner: AbstractCreature, @Suppress("UNUSED_PARAMETER") amount: Int) : BasePower(POWER_ID) {
    init {
        setOwnerAmount(owner, -1)
        type = PowerType.BUFF
        isTurnBased = true
        loadRegion("anger")
        updateDescription()
    }

    override fun atEndOfTurn(isPlayer: Boolean) {
        addToBot(iop().changeStance(owner, StanceAloft.STANCE_ID))
        addToBot(RemoveSpecificPowerAction(owner, owner, this))
    }

    companion object {
        @JvmField val POWER_ID = GrackleMod.makeId(SwoopPower::class.java)
    }
}